test/test_hash_table.pf

```diff
@@ -0,0 +1,33 @@
+@test
+subroutine test_hash_table_add_get()
```

Comment:
Can we add comments about what each of these tests is "testing"? That way someone can quickly see what's covered.

Reply:
Updated test names and added comment around assert.  I'm trying to think of a human-readable way of expressing what the test is doing but the API used in the hash table and hashable is making this a little tricky so just let me know if this doesn't sound as clear or as explicit as it could be and I'll take another crack at it.